src/calendar.rs

```diff
@@ -73,7 +73,7 @@ impl Calendar {
     }
 
     /// Writes `Calendar` into a `Writer` using `std::fmt`.
-    fn fmt_write<W: fmt::Write>(&self, out: &mut W) -> Result<(), fmt::Error> {
+    pub fn fmt_write<W: fmt::Write>(&self, out: &mut W) -> Result<(), fmt::Error> {
```

Comment:
actually this is not meant to be public since as you mentioned it's not the prefered way of formatting. It's only used for `fn print` and the `ToString` implementation. So I'd prefer to keep it private.

Reply:
I see, you're right, I'll take the last commit out of the PR.

I was thinking about this more - it would be best to implement Display trait instead. We would get to_string() for free and also resolve the duplication TODO in code. I may submit another PR for this.